tests/qir_test.py

```diff
@@ -118,6 +117,126 @@ def test_wasm_only(self) -> None:
+        assert barriers[2].op.type == OpType.Barrier
+        assert (
+            barriers[2].op.data
+            == '{"name": "__quantum__rt__result_record_output", "arg": 0}'
+        )
+        assert barriers[2].qubits == []
+        assert barriers[2].bits[0].index[0] == 0
+        assert barriers[3].op.type == OpType.Barrier
+        assert (
+            barriers[3].op.data
+            == '{"name": "__quantum__rt__result_record_output", "arg": 1}'
+        )
+        assert barriers[3].qubits == []
+        assert barriers[3].bits[0].index[0] == 1
+
+    def test_tagged_rt_functions(self) -> None:
+        rt_function_file_path = qir_files_dir / "tagged_rt_functions.bc"
+        circuit = circuit_from_qir(rt_function_file_path)
+
+        barriers = circuit.commands_of_type(OpType.Barrier)
+
+        assert barriers[0].op.type == OpType.Barrier
+        assert (
+            barriers[0].op.data == '{"name": "__quantum__rt__integer_record_output",'
+            ' "arg": "%0", "tag": "0_t2\\u0000"}'
+        )
+        assert barriers[0].qubits == []
+        assert circuit.get_c_register(
+            barriers[0].args[0].reg_name
+        ) == circuit.get_c_register("%0")
+        assert barriers[1].op.type == OpType.Barrier
+        assert (
+            barriers[1].op.data == '{"name": "__quantum__rt__bool_record_output",'
+            ' "arg": "%1", "tag": "0_t3\\u0000"}'
+        )
+        assert barriers[1].qubits == []
+        assert circuit.get_c_register(
+            barriers[1].args[0].reg_name
+        ) == circuit.get_c_register("%1")
+        assert barriers[2].op.type == OpType.Barrier
+        assert (
+            barriers[2].op.data == '{"name": "__quantum__rt__result_record_output",'
+            ' "arg": 0, "tag": "0_t0\\u0000"}'
+        )
+        assert barriers[2].qubits == []
+        assert barriers[2].bits[0].index[0] == 0
+        assert barriers[3].op.type == OpType.Barrier
+        assert (
+            barriers[3].op.data == '{"name": "__quantum__rt__result_record_output",'
+            ' "arg": 1, "tag": "0_t1\\u0000"}'
+        )
+        assert barriers[3].qubits == []
+        assert barriers[3].bits[0].index[0] == 1
+
+    def test_select(self) -> None:
+        select_function_file_path = qir_files_dir / "select.bc"
+
+        circuit = circuit_from_qir(select_function_file_path)
+        output_register = circuit.get_c_register("%1")
+
+        conditionals = []
+
+        for com in circuit.get_commands():
+            if (
+                com.op.type == OpType.Conditional
+            ):  # Raising an error if I use .commands_of_type(OpType.Conditional)
+                conditionals.append(com)
+
+        conditional0 = conditionals[0]
+        assert conditional0.op.type == OpType.Conditional
+        assert conditional0.bits == list(output_register)
+        assert conditional0.op.value == 1
+        assert (
+            sum([int(n) * 2**k for k, n in enumerate(conditional0.op.op.values)])
```

Comment:
Why is the cast to `int` necessary?

Reply:
Reminiscent of the old debugging times. Removed.